engine-src/Game/LambdaHack/Client/UI/FrameM.hs

```diff
@@ -169,18 +170,20 @@ promptGetKey dm ovs onBlank frontKeyKeys = do
   -- to also capture choice of items from menus, etc.
   -- Notice that keys coming from macros (in-game, content, config)
   -- are recorded as well and this is well defined and essential.
-  CCUI{coinput=InputContent{brevMap}} <- getsSession sccui
+  CCUI{coinput} <- getsSession sccui
   modifySession $ \sess ->
     sess { sdisplayNeeded = False
          , sturnDisplayed = True
-         , sactionPending = addToMacro brevMap km $ sactionPending sess }
+         , sactionPending = addToMacro coinput km $ sactionPending sess }
   return km
 
-addToMacro :: M.Map HumanCmd.HumanCmd [K.KM] -> K.KM -> [ActionBuffer]
-           -> [ActionBuffer]
+addToMacro :: InputContent -> K.KM -> [ActionBuffer] -> [ActionBuffer]
 addToMacro _ _ [] = error "cannot add macro to empty action stack"
-addToMacro brevMap km (abuff : abuffs) =
-  let newBuffer = case M.lookup HumanCmd.Record brevMap of
+addToMacro InputContent{bcmdMap, brevMap} km (abuff : abuffs) =
+  let mRecordLast = case (\(_, _, cmd) -> cmd) <$> M.lookup km bcmdMap of
```

Comment:
This is not type correct currently, but indeed using `bcmdMap` is much better here than using the inverse function `brevMap`. Please remove 'brevMap` everywhere in this file in other files, whenever it appears in this role.